src/Module/Upgrade/Upgrade1_5_0.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gett\MyparcelBE\Module\Upgrade;
+
+use Gett\MyparcelBE\Constant;
+use Gett\MyparcelBE\Database\Table;
+use MyParcelNL\Sdk\src\Support\Collection;
+
+class Upgrade1_5_0 extends AbstractUpgrade
+{
+    private $carrierConfigurationTable;
+
+    /**
+     * @throws \PrestaShopDatabaseException
+     */
+    public function upgrade(): void
+    {
+        $this->addInsuranceOptionsForCarriers();
+    }
+
+    /**
+     * Adds insurance_from_price and insurance_max_amount to carrier configuration if not already present
+     *
+     * @throws \PrestaShopDatabaseException
+     */
+    private function addInsuranceOptionsForCarriers(): void
+    {
+        $this->carrierConfigurationTable = Table::withPrefix(Table::TABLE_CARRIER_CONFIGURATION);
+        $configName                      = Constant::INSURANCE_CONFIGURATION_NAME;
+
+        $query   = <<<SQL
+SELECT DISTINCT id_carrier, name, value FROM $this->carrierConfigurationTable WHERE name LIKE '%$configName'
+SQL;
+        $records = new Collection($this->db->executeS($query));
+
+        foreach ($records as $record) {
+            $carrierId = (int) $record['id_carrier'];
```

Comment:
Ik ben een klein beetje getraumatiseerd door dingen die later ineens niet meer bestaan en die dan oude migraties slopen, zou je dit geheel zo alles-safe mogelijk kunnen maken? Bijvoorbeeld, er vanuitgaan dat `$record` misschien geen `id_carrier` heeft?

Reply:
Dat kan niet want die wordt geselecteerd door de query, dus zal altijd aanwezig zijn in de record, tenzij de query heeft gefaald. Dit is een tabel die wij zelf beheren, met carrier settings erin. Ik denk dat rekening houden met een gewijzigde structuur van die tabel tijdens een migratie (van een versie waarin de structuur bekend is) onder overengineering valt, ook al begrijp ik je trauma geheel.